### DISCUSSION TITLE
Run default code if no subcommand is used

### DISCUSSION THREAD

**Discussion by sigmaSd** [c4spar/cliffy#537] [Q&A] [answered]
```ts
    const cliffy = await new Command()
      .name("chef")
      .command("run", "run a binary")
      .action()
      .parse(Deno.args);
```

I want to show the default help menu, if the binary is run with no sub-commands

so far I only found this hack
```ts
if (cliffy.cmd._name === "chef") cliffy.cmd.showHelp()
```

**Comment by c4spar** [comment, marked as answer]
You can use the `.showHelp()` or `.getHelp()` method inside the action handler. Create a reference of your command or use `this` to get the current command instance.

```ts
.action(function () {
  this.showHelp();
});
```

**Reply by sigmaSd** [reply]
but I want to show help if no action handler gets triggered

**Reply by c4spar** [reply]
You need to use the action handler of the main command. If no sub-command matches, the action handler of the main command is executed:

```ts
await new Command()
  .name("chef")
  .action(function() {
    this.showHelp();
  })
  .command("run", "run a binary")
  .action(() => console.log("run action"))
  .parse(Deno.args);
```

**Reply by sigmaSd** [reply]
nice thanks

**Reply by justinmchase** [reply]
In case you come here like me wondering how you do this if you just have a bunch of top level commands and no sub-commands, the above still works but I had to wrap my brain around the concept that the top level command is just the name of the executable essentially, it doesn't require explicit arguments. Or in other words it is the _default command_ if no arguments are provided, the rest are sub-command which do require arguments.

```ts
await new Command()
  .name("example").action(...) // invoked if full command is `deno run main.ts`
  .command("a").action(...) // invoked if command is `deno run main.ts a`
  .command("b").action(...) // invoked if command is `deno run main.ts b`
  .parse(Deno.args)
```
